zeebe/exporter-common/src/main/java/io/camunda/zeebe/exporter/common/auditlog/transformers/AuditLogTransformer.java

```diff
@@ -22,11 +26,38 @@
  *
  * @param <R> the record value type for this transformer
  */
-public interface AuditLogTransformer<R extends RecordValue, T> {
+public interface AuditLogTransformer<R extends RecordValue> {
+
+  public static final Logger LOG = LoggerFactory.getLogger(AuditLogTransformer.class);
 
   TransformerConfig config();
 
-  default void transform(final Record<R> record, final T entity) {}
+  default AuditLogEntry create(final Record<R> record) {
+    final AuditLogEntry log = AuditLogEntry.of(record);
+
+    try {
+      transform(record, log);
+    } catch (final Exception e) {
+      LOG.error(
+          "Error transforming audit log entity for record with key {}: {}",
+          record.getKey(),
+          e.getMessage(),
+          e);
+    }
+
+    if (log.getResult() == null) {
+      if (RecordType.COMMAND_REJECTION.equals(record.getRecordType())) {
+        // TODO: set rejection type and reason to AuditLogEntity#details
```

Comment:
🔧 We can remove the `TODO` from here; the serialization of the operation details (including the rejection message) will need to be moved to a different place now.